File: mono/src/symbol/enumeration.rs
use serde::{Serialize, Deserialize};

use crate::symbol::Declaration;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Entry {
	#[serde(flatten)]
	pub declaration: Declaration,

	/// Value that are explicitly set in code expressions
	pub value: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Enumeration {
    #[serde(flatten)]
	pub declaration: Declaration,

	pub entries: Vec<Entry>,
}

File: typings/src/interfaces/symbol/enumeration.ts
import { Declaration } from './base';

export interface Entry extends Declaration {
    /**
     * @brief Value that are explicitly set in code expressions
     */
    value?: string,
}

export interface Enumeration extends Declaration {
    /**
     * @brief Enum entries
     * @readonly
     */
    readonly entries: Entry[],
}
